src/SDiagonal.jl

```diff
@@ -0,0 +1,117 @@
+# Originally contributed by D. Getz (https://github.com/getzdan), M. Schauer
+# at https://github.com/mschauer/Bridge.jl under MIT License
+
+import Base: getindex,setindex!,==,-,+,*,/,\,transpose,ctranspose,convert, size, abs, real, imag, conj, eye, inv
+import Base.LinAlg: ishermitian, issymmetric, isposdef, factorize, diag, trace, det, logdet, expm, logm, sqrtm
+
+@generated function scalem{T, M, N}(a::SMatrix{M,N, T}, b::SVector{N, T})
```

Comment:
use `StaticMatrix` and `StaticVector` not `SMatrix` and `SVector` in the signature, please. Same with the method below.